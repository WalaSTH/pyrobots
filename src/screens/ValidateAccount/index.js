import { useEffect, useState } from "react";
import {
  Container,
  Card,
  Typography,
  CircularProgress,
  Grid,
  Button,
} from "@mui/material";
import axios from "axios";
import { useNavigate } from "react-router-dom";
import { useSearchParams } from "react-router-dom";

const endpoint = "http://127.0.0.1:8000/validate_account";

export default function ValidateAccount() {
  const navigate = useNavigate();

  // Fetch query params
  const [searchParams] = useSearchParams();
  const token = searchParams.get("token");

  // Check if token is expired
  const [isValidating, setIsValidating] = useState(true);
  const [isTokenValid, setIsTokenValid] = useState(true);

  useEffect(() => {
    axios
      .get(endpoint, {
        params: {
          token: token,
        },
      })
      .then(function () {
        setIsValidating(false);
        navigate("/login");
      })
      .catch(function () {
        setIsValidating(false);
        setIsTokenValid(false);
      });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <Container
      component="main"
      sx={{
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
      }}
    >
      {isValidating && (
        <Card
          variant="outlined"
          sx={{
            marginTop: 3,
            padding: 3,
            borderRadius: 3,
            textAlign: "center",
          }}
        >
          <CircularProgress />
          <Typography>Validating account</Typography>
        </Card>
      )}
      {!isTokenValid && (
        <Card
          variant="outlined"
          sx={{
            marginTop: 3,
            padding: 3,
            borderRadius: 3,
            textAlign: "center",
          }}
        >
          <Grid container spacing={2}>
            <Grid item xs={12}>
              <Typography component="h1" variant="h5" children="Link Expired" />
            </Grid>

            <Grid item xs={12}>
              <Button
                variant="contained"
                color="primary"
                size="large"
                onClick={() => {
                  navigate("/login");
                }}
              >
                Back to login
              </Button>
            </Grid>
          </Grid>
        </Card>
      )}
    </Container>
  );
}
