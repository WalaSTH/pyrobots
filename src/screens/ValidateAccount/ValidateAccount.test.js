import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import ValidateAccount from "./";
import { BrowserRouter } from "react-router-dom";

describe("<ResetPassword", () => {
  it("component rendering", async () => {
    const component = renderer.create(
      <BrowserRouter>
        <ValidateAccount />
      </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
