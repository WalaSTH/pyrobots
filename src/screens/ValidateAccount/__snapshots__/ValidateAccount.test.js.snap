// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPassword component rendering 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthLg css-12hdurq-MuiContainer-root"
>
  <div
    className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-he3bwl-MuiPaper-root-MuiCard-root"
  >
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style={
        {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={{}}
        />
      </svg>
    </span>
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Validating account
    </p>
  </div>
</main>
`;
